networkx/algorithms/coloring/edge_coloring.py

```diff
@@ -0,0 +1,147 @@
+import networkx as nx
+
+__all__ = ["edge_coloring"]
+
+
+@nx.utils.not_implemented_for("directed")
+@nx.utils.not_implemented_for("multigraph")
+@nx._dispatchable
+def edge_coloring(G):
+    """This function performs edge coloring on a given graph G.
+
+    Parameters
+    ----------
+    G: NetworkX graph object representing the input graph.
+
+    Returns
+    -------
+    coloring: A dictionary where keys are tuples representing edges of the graph,
+                and values are integers representing colors assigned to the edges.
+
+    Edge coloring is the assignment of colors to the edges of a graph in such a way
+    that no two adjacent edges share the same color.
+
+    References
+    ----------
+    This implementation follows the proof of the lemma given in Zdenek Dvorak's lecture notes
+    titled Graph coloring: Heawood and Brooks theorems, edge coloring which states
+    Let G be a graph of maximum degree at most d with edges properly colored using
+    d + 1 colors. Let u and v be distinct non-adjacent vertices of G.
+    Then there exists a proper edge coloring using d + 1 colors
+    such that the same color missing at both u and v.
+    https://iuuk.mff.cuni.cz/~rakdver/kgii/lesson20-6.pdf
```

Comment:
Can you make a References section with the full reference to this paper. Links rarely last more than a few years (except for the DOI link to the article). Full reference is more likely to last without additional maintenance.

Reply:
We are actually not able to find the DOI link since our reference is a lecture note.